components/lib/inputotp/InputOtp.js

```diff
@@ -0,0 +1,182 @@
+import React, { useRef, useContext, useState } from 'react'
+import { useMergeProps } from '../hooks/Hooks';
+import { PrimeReactContext } from '../api/Api';
+import { InputOtpBase } from './BaseInputOtp';
+
+export const InputOtp = React.memo(
+    React.forwardRef((inProps, ref) => {
+	const elementRef = useRef(ref);
+	const mergeProps = useMergeProps();
+	const context = useContext(PrimeReactContext);
+	const props = InputOtpBase.getProps(inProps, context);
+	const { ptm, cx } = InputOtpBase.setMetaData({
+		props,
+		...props.__parentMetadata,
+	});
+
+	const defaultValue = props.value ? props.value?.toString()?.split?.('') : new Array(props.length);
+	const [tokens, setTokens] = useState(defaultValue);
+
+	const moveToNextInput = (event) => {
+		const nextInput = event.target.nextElementSibling;
+
+		if (!nextInput) return;
+
+		if (nextInput.nodeName === 'INPUT') {
+			nextInput.focus();
+			nextInput.select();
+		}
+	}
+
+	const moveToPrevInput = (event) => {
+		const prevInput = event.target.previousElementSibling;
+
+		if (!prevInput) return;
+
+		if (prevInput.nodeName === 'INPUT') {
+			prevInput.focus();
+			prevInput.select();
+		}
+	}
+
+	const onChange = (event, value) => {
+		props?.onChange?.({
+			originalEvent: event,
+			value: value.join('')
+		})
+	}
+
+	const updateTokens = (event, index) => {
+		const inputValue = event.target.value;
+		const newTokens = [...tokens];
+		newTokens[index] = inputValue;
+		newTokens = newTokens.join('');
+		newTokens = newTokens ? newTokens.split('') : new Array(props.length);
+
+		setTokens(newTokens);
+		onChange(event, newTokens);
+	}
+
+	const onInput = (event,index) => {
+		updateTokens(event, index);
+		
+
+		if (event.nativeEvent.inputType === 'deleteContentBackward') {
+			moveToPrevInput(event);
+		} else if (event.nativeEvent.inputType === 'insertText' || event.nativeEvent.inputType === 'deleteContentForward') {
+			moveToNextInput(event);
+		}
+	}
+
+	const onPaste = (event) => {
+		let paste = event.clipboardData.getData('text');
+
+		if (paste.lenght) {
+			let pastedCode = paste.substring(0, props.length + 1);
+
+			if (!props.integerOnly || !isNaN(pastedCode)) {
+				const newTokens = pastedCode.split('');
+
+				setTokens(newTokens);
+				onChange(event, newTokens);
+			}
+		}
+	}
+
+	const onFocus = (event) => {
+		event.target.select();
+		props?.focus?.(event);
+	}
+
+	const onBlur = (event) => {
+		props?.blur?.(event);
+	}
+
+	const onKeydown = (event) => {
+		switch (event.code) {
+			case 'ArrowLeft': {
+				moveToPrevInput(event);
+				event.preventDefault();
+				break;
+			}
+
+			case 'ArrowRight': {
+				moveToNextInput(event);
+				event.preventDefault();
+				break;
+			}
+
+			case 'Backspace': {
```

Comment:
In PrimeFaces we made the `Backspace` and `Delete` key do the same thing as its expected behavior for most users.

Reply:
I basically inherited this function from PrimeVue InputOtp